### PR TITLE
03 DynamoSubpopulationDaoTest, DynamoUploadDaoMockTest, DynamoSurveyDaoTest

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        assertEquals(result.size(), 1);
+        assertFalse(result.get(0).isDefaultGroup());
+    }
+    
+    @Test
+    public void getSubpopulationsCreateDefaultDoesNotIfThereAreExistingSubpops() {
+        List<DynamoSubpopulation> subpopList = ImmutableList.of();
+        when(mockMapper.query(eq(DynamoSubpopulation.class), any())).thenReturn(mockQueryList);
+        when(mockQueryList.isEmpty()).thenReturn(subpopList.isEmpty());
+        when(mockQueryList.stream()).thenReturn(subpopList.stream());
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java:191]: I think this test and the previous have switched names.

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+import org.testng.annotations.Test;
+
+import org.sagebionetworks.bridge.dao.CriteriaDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.Criteria;
+import org.sagebionetworks.bridge.models.subpopulations.Subpopulation;
+import org.sagebionetworks.bridge.models.subpopulations.SubpopulationGuid;
+
+public class DynamoSubpopulationDaoTest extends Mockito {
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java:39]: Need tests for

* getSubpopulations() with createDefault = false
* updateSubpopulation cannot flip the isDefault flag

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        subpop.setStudyIdentifier(TEST_STUDY_IDENTIFIER);
+        
+        dao.createSubpopulation(subpop);
+        
+        verify(mockMapper).save(subpopCaptor.capture());
+        Subpopulation savedSubpop = subpopCaptor.getValue();
+        assertNotNull(savedSubpop.getGuidString());
+        assertFalse(savedSubpop.isDeleted());
+        assertNull(savedSubpop.getVersion());
+        assertFalse(savedSubpop.isDefaultGroup());
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java:91]: isDeleted() and isDefaultGroup() should be set to true to verify that we overwrite them in createSubpopulation().

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoUploadDaoMockTest.java [hunk 4]
```diff
@@ -447,6 +469,44 @@ public void getStudyUploadsBadOffsetKey() {
...
+    public void getStudyUploadsMaxPageSizeEnforced() { 
+        StudyIdentifier studyId = new StudyIdentifierImpl("test-study");
+        DateTime startTime = DateTime.now().minusDays(4);
+        DateTime endTime = DateTime.now();
+        
+        dao.getStudyUploads(studyId, startTime, endTime, 101, null);
+    }
+    
+    @Test
+    public void deleteUploadsForHealthCode() {
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoUploadDaoMockTest.java:489]: How about an equivalent to deleteByHealthCodeSilentlyFails() (for when there are no uploads for the given healthcode)?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]
~~Is DynamoUploadDedupeDaoTest also supposed to be in this pull request?~~ Disregard. Looks like it got bundled with one of the service test PRs.

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java:191]: Yes, switched and improved the names.

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java:39]: added test and additional testing to updateSubpopulation() test.

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java:91]: Added all the defaults as different values and now verifying they are all in their default state when saved.

File: src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoUploadDaoMockTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoUploadDaoMockTest.java:489]: Added

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**